test/promise-subclass.js

```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = (tap) => {
+  return class SubclassedPromise extends Promise {
+    then(onSuccess, onReject) {
+      tap.type(onSuccess, 'function');
+      tap.type(onReject, 'undefined');
+      return Promise.prototype.then.call(this, onSuccess, onReject);
```

Comment:
Any reason not to just do super.then(...) here?

Reply:
Not that I can think of, I've updated it.